### DISCUSSION TITLE
Why a different helper binary for libfuse3?

### DISCUSSION THREAD

**Discussion by mgord9518** [libfuse/libfuse#1010] [General]
What's the reasoning behind having a `fusermount3`? It appears to be fully compatible with `fusermount` to the point where certain distros (Debian) even symlink it for backwards-compatibility purposes 

I understand that libfuse2 and libfuse3 are meant to be co-installable, but could they not share the same SUID helper?

**Comment by ashleypittman** [comment]
libfuse2 and libfuse3 are meant to be co-installable but they're also meant to be independent so it should be possible to use libfuse3 without a libfuse2 install.

**Comment by mgord9518** [comment]
So is it mainly about making it easy for distros to package? In that case would it make sense for a distro to have fusermount as a seperate package that both libfuse 2 and 3 rely on?

**Comment by bsbernd** [comment]
fusermount is supposed to be bound to a libfuse version/release. So far it has been compatible, but at some point that won't hold. Example, current fusermount supports to pass the fd number as argument instead of an env var. We just don't use that by default yet. I don't remember which discussion this was in, but there was already a request where using the parameter would have helped.

**Comment by probonopd** [comment]
> fusermount is supposed to be bound to a libfuse version/release. So far it has been compatible, but at some point that won't hold. 

It would help a tremendous lot to know for sure that future versions of `fusermount` will continue to work with all versions of libfuse ("backward compatibility").

One might say, "just install an older version of `fusermount` in parallel", but that is not what distributions are doing. Ubuntu, for example, dropped `fusermount` from one ISO to the next in favor of `fusermount3`, with no transitional time period. This has proven to be a BIG hassle for the AppImage file format, which relies on `fusermount`.

Please also see the related discussion:
* https://github.com/libfuse/libfuse/discussions/1007
